system/Throttle/Throttler.php

```diff
@@ -128,12 +126,21 @@ public function check(string $key, int $capacity, int $seconds, int $cost = 1):
         // If $tokens >= 1, then we are safe to perform the action, but
         // we need to decrement the number of available tokens.
         if ($tokens >= 1) {
-            $this->cache->save($tokenName, $tokens - $cost, $seconds);
-            $this->cache->save($tokenName . 'Time', time(), $seconds);
+            $tokens = $tokens - $cost;
+            $this->cache->save($tokenName, $tokens, $seconds);
+            $this->cache->save($tokenName . 'Time', $this->time(), $seconds);
+
+            $this->tokenTime = 0;
 
             return true;
         }
 
+        // How many seconds till a new token is available.
+        // We must have a minimum wait of 1 second for a new token.
```

Comment:
Is this a new minimum? If so this should be noted in the user guide and changelog. Is there any reason we would want to support fractional seconds, make this whole thing millisecond based?

Reply:
No, it is not new. And it is already documented:
> In this case, the minimum enforced wait time is one second.
https://codeigniter4.github.io/CodeIgniter4/libraries/throttler.html#getTokentime